test/main/modules.js

```diff
@@ -419,4 +419,35 @@ QUnit.module( "QUnit.module", function() {
 			assert.expect( 9 );
 		} );
 	} );
+
+	QUnit.module( "Improperly invoked hooks", function( hooks ) {
+		QUnit.module( "Nested module with different hooks variable name", function( innerHooks ) {
+			this.outerHookRan = false;
+			try {
+				hooks.beforeEach( function() {
+					this.outerHookRan = true;
+				} );
+			} catch ( e ) {
+				this.beforeEachError = e;
+			}
+
+			this.innerHookRan = false;
+			innerHooks.beforeEach( function() {
+				this.innerHookRan = true;
+			} );
+
+			QUnit.test( "calling parent module's beforeEach errors", function( assert ) {
+				assert.strictEqual( this.beforeEachError.message,
```

Comment:
I believe this test will fail in a less-than-ideal way if the error is not thrown. It would abort from property `message` on undefined `this.beforeEachError`, instead of on a failed assertion.

Perhaps an approach could be to have the `catch` statement above assign the result of `e.message`. That way over here you have a consistently accesible value that is either undefined or string.

Reply:
Updated tests, now running the new tests without the change to src/module.js will fail with:

```
>> Testing test/main/modules.js
>>
>> test: calling parent module's beforeEach errors
>> module: Nested module with different hooks variable name
>> message: Correct error thrown by beforeEach
>>   at Object.<anonymous> (/Users/rcohen/code/oss/forks/qunit/test/main/modules.js:442:12)
>>
>> test: calling parent module's beforeEach errors
>> module: Nested module with different hooks variable name
>> message: outer hook beforeEach should not run
>>   at Object.<anonymous> (/Users/rcohen/code/oss/forks/qunit/test/main/modules.js:445:17)
>>
>> test: hooks error when invoked during test execution
>> module: Improperly invoked hooks
>> message: Correct error thrown by beforeEach
>>   at Object.<anonymous> (/Users/rcohen/code/oss/forks/qunit/test/main/modules.js:451:11)
>> 35 tests in 19ms, 30 passed, 2 failed, 3 skipped.
```